app/assets/javascripts/application.js

```diff
@@ -375,7 +375,7 @@ $(function() {
 
 function update_puppetclasses(element) {
   var host_id = $("form").data('id')
-  var env_id = $('*[id*=environment_id]').val();
+  var env_id = $('*[name*=environment_id]').val();
```

Comment:
normally, this JS expects an ID and not a name, if the server is smart enough to figure out a name instead of an idea is a sideeffect but you are changing behaviour.

IMHO, find the right selector to give you back the id would not generate any unexpected results, name might.